Clean up

icemac opened the pull request:
Most important is the first commit which no longer uses absolute paths which newer `setuptools` do not like any more. But this only seems to happen under some circumstances at lest not when running the buildout of the package itself.

Commit:
Ensure package can be installed.

Newer setuptools versions do not like that an absolute path is used.

Commit:
Ignore files and folder created by virtualenv and sort lines.

Commit:
Rename so the extension matches the file contents.

Commit:
Clean-up.

tlotze commented on setup.py:
```diff
@@ -11,12 +11,11 @@
-root_files = glob.glob(project_path("*.txt"))
+root_files = glob.glob("*.txt")
+root_files.extend(glob.glob("*.rst"))
 data_files = [("", [name for name in root_files
```
Das Ziel Deiner Änderungen bzgl. `project_path` war ja, absolute Pfade aus den `data_files` wegzubekommen. Ich würde es besser finden, alle Lesezugriffe auf `rst`-Dateien und alles Globbing wie bisher mit absoluten Pfaden durchzuführen und die Verzeichnisse erst beim Zusammensetzen der `data_files` abzuschneiden. Vielleicht weißt Du ja von Garantien für die Ausführung von `setup.py`, aber ohne jetzt viel ausprobiert zu haben, würde ich erst mal nicht ausschließen wollen, daß das Arbeitsverzeichnis je nach Installationswerkzeug auch mal ein anderes als das Projektverzeichnis sein kann. Wenn das der Fall ist, dann produziert `setup.py` jetzt mit bloßen Dateinamen jedenfalls Fehler, sobald es z.B. kein `README.rst` finden kann.

icemac replied:
Ich habe vielleicht 10 gocept-Pakete angepasst und dort überall die Verwendung von absoluten Pfaden entfernt, danach hat es in keinem Fall Probleme gegeben. In der Regel wird ein Paket ja entweder mit `pip` oder `buildout` installiert und die kommen mit relativen Pfaden klar.

Wenn man in einem `virtualenv` versucht, `tl.eggdeps` zu installieren, wird – neben der Fehlermeldung – folgender Text angezeigt:

>   setup() arguments must *always* be /-separated paths relative to the
>  setup.py directory, *never* absolute paths.

Das war für mich der Grund, alle absoluten Pfade rauszureißen.

vernans replied:
@tlotze Auch bei mir trat der Fehler auf, die Installation durch buildout war ohne macs Branch unmöglich.

tlotze commented on setup.py:
```diff
@@ -44,20 +43,22 @@
     "Intended Audience :: Developers",
     "Intended Audience :: System Administrators",
     "License :: OSI Approved :: Zope Public License",
-    "Programming Language :: Python",
```
Was spricht gegen den einfachen "Python"-Klassifikator? (Und ggf. auch "Python :: 2".)

icemac replied:
Lediglich, dass er doppelt ist. Die spezifischen Klassifikatoren schließen die allgemeineren ein. – So zumindest die Praxis des alten PyPI, Warehouse macht es anscheinend wieder anders, dass man doch wieder alle allgemeinen Klassifikatoren angegeben muss 😢 
Siehe: https://pypi.org/search/?q=icemac.truncatetext&o=&c=Programming+Language+%3A%3A+Python wo das gesuchte Paket nicht gefunden wird, weil es "Programming Language :: Python" nicht als Klassifikator hat.

tlotze commented on setup.py:
```diff
@@ -44,20 +43,22 @@
       classifiers=classifiers,
       author="Thomas Lotze",
       author_email="thomas@thomas-lotze.de",
-      url="http://thomas-lotze.de/en/software/eggdeps/",
```
Hier würde ich gern die alte URL beibehalten, zumindest bis ich mir mal angesehen habe, ob ich eine Beispielgrafik auf der Projektseite im Github einbinden kann und wie das ggf. am sinnvollsten geht. Gibt's eigentlich mehr oder weniger wohldefinierte Richtlinien dafür, was als URL üblich oder gemeinhin akzeptiert ist?

icemac replied:
In den Zopefoundation- und gocept-Repos verwenden wir immer die Repos-URL, weil man vom Repos aus alle relevanten Informationen finden kann und diese immer auf dem aktuellen Stand sind.

icemac commented:
@tlotze Was muss noch getan werden, damit dieser PR gemergt und veröffentlicht werden kann. Wie @vernans geschrieben hat, lässt sich das Paket derzeit nicht neu installieren.

tlotze committed:
add back unversioned Python classifier

tlotze committed:
don't include data files in package at all

Including data files might have been OK with zc.buildout which keeps installed
eggs to their own directories, but when using pip, all the root data files are
put in the venv root which just isn't an option.

tlotze committed:
update changelog
